app/models/llms/basic_workflow/prompting.py

```diff
@@ -0,0 +1,189 @@
+from pathlib import Path
+from typing import Dict, Any, Optional
+
+
+class PromptManager:
+    """
+    Manages prompt templates for the Voyager T800 travel planning system.
+    
+    This class handles loading, formatting, and managing prompt templates from
+    text files. It provides robust error handling for missing files and template
+    variables, with intelligent fallback values for common template placeholders.
+    
+    Attributes:
+        prompts_dir (Path): Directory containing prompt template files
+    """
+    
+    def __init__(self, prompts_dir: Optional[str] = None):
+        """
+        Initialize the PromptManager with a prompts directory.
+        
+        Args:
+            prompts_dir: Optional path to the prompts directory. If None, uses
+                        the default prompts directory relative to this file.
+        """
+        if prompts_dir is None: 
+            current_file_dir = Path(__file__).parent
+            self.prompts_dir = (current_file_dir.parent.parent / "prompts").resolve()
```

Comment:
- The logic that sets the default path for prompts_dir may fail if the directory structure is changed. Consider making the default path a constant or configuration variable.
- The path calculation is a bit obscure - add a comment explaining the intent (e.g., "default: .../app/prompts").
- If the prompts directory does not exist, consider raising an error or creating it, depending on use-case.